build/gulp/tasks/perf.ts

```diff
@@ -117,6 +117,43 @@ task('perf:build', cb => {
   webpackPlugin(require('../../../build/webpack.config.perf').default, cb)
 })
 
+task('perf:run-memory', async () => {
+  const browser = await puppeteer.launch(
+    safeLaunchOptions(
+      process.env.CI && {
+        executablePath: 'google-chrome-unstable',
+      },
+    ),
+  )
+  const context = await browser.createIncognitoBrowserContext()
+  const page = await context.newPage()
+
+  const client = await page.target().createCDPSession()
+  await client.send('HeapProfiler.enable')
+
+  const chunks: string[] = []
+
+  client.on('HeapProfiler.addHeapSnapshotChunk', ({ chunk }) => {
+    chunks.push(chunk)
+  })
+
+  await page.goto(`http://${config.server_host}:${config.perf_port}`)
+
+  await page.evaluate(() => window.runMeasures())
+  await page.evaluate(() => gc()) // collect garbage manually
```

Comment:
Triggering garbage collection manually, need to be checked again

https://media-codings.com/articles/automatically-detect-memory-leaks-with-puppeteer